command/meta.go

```diff
@@ -179,3 +182,125 @@ func (m *Meta) GetConfigFromJSON(cla *MetaArgs) (packer.Handler, int) {
 	}
 	return core, ret
 }
+
+func (m *Meta) DetectBundledPlugins(handler packer.Handler) hcl.Diagnostics {
+	var plugins []string
+
+	switch h := handler.(type) {
+	case *packer.Core:
+		plugins = m.detectBundledPluginsJSON(h)
+	case *hcl2template.PackerConfig:
+		// enable packer to start plugins requested in required_plugins.
+		diags := h.DetectPluginBinaries()
+		if diags.HasErrors() {
+			return diags
+		}
+
+		plugins = m.detectBundledPluginsHCL2(handler.(*hcl2template.PackerConfig))
+	}
+
+	if len(plugins) == 0 {
+		return nil
+	}
+
+	buf := &strings.Builder{}
+	buf.WriteString("This configuration uses some bundled plugins, which is a feature that will be removed from Packer in an upcoming version.\n\n")
+	switch handler.(type) {
+	case *packer.Core:
+		buf.WriteString("Plase consider installing those plugins with the 'packer plugins install' command\n\n")
+
+		for _, plugin := range plugins {
+			fmt.Fprintf(buf, "* packer plugins install %s\n", plugin)
+		}
+
+		buf.WriteString("\nAlternatively, if you change your templates to HCL2, you may use 'packer init' with a `required_plugins` block in your template.\n\n")
+	}
+
+	buf.WriteString("In HCL2, consider adding the following section to your template:\n")
+	buf.WriteString(generateRequiredPluginsBlock(plugins))
+	buf.WriteString("\nThen run 'packer init' to manage installation of the plugins")
+
+	return hcl.Diagnostics{
+		&hcl.Diagnostic{
+			Severity: hcl.DiagWarning,
+			Summary:  "Bundled plugins used",
+			Detail:   buf.String(),
+		},
+	}
+}
+
+func (m *Meta) detectBundledPluginsJSON(core *packer.Core) []string {
+	log.Printf("[PLUGIN] Running DetectBundledPluginsJSON")
+
+	bundledPlugins := map[string]struct{}{}
+
+	tmpl := core.Template
+	if tmpl == nil {
+		panic("No template parsed already")
```

Comment:
Oh yes definitely. I would presume we'll hit that crash if it ever happens before we roll a new version out, but it can't hurt to point out that this bug's on us. I'll amend this.